server/multitenancy/tenant_resolver.ts

```diff
@@ -143,7 +143,18 @@ function resolve(
     return PRIVATE_TENANT_SYMBOL;
   }
 
-  // fall back to the first tenant in the available tenants
+  /**
+   * fall back to the first tenant in the available tenants
+   * Under the condition of enabling multitenancy, if the user has disabled both 'Global' and 'Private' tenants:
+   * it will remove the default global tenant key for custom tenant.
+   */
+  if (
+    Object.keys(availableTenantsClone).length > 1 &&
+    availableTenantsClone.hasOwnProperty(globalTenantName)
+  ) {
+    delete availableTenantsClone[globalTenantName];
+    return findKey(availableTenantsClone, () => true);
```

Comment:
The return here isn't necessary because it will hit the same return statement when it exits the block.

Reply:
Nice catch @cwperks! I will update this with the test case.